### ISSUE TITLE
Undefined method `serialize_to_hash` on Serializer page

### ISSUE THREAD

**Issue by CodingItWrong** [cerebris/jsonapi-resources-site#53]
`guide/serializer.html` suggests that we can do:

```
post = Post.find(1)
JSONAPI::ResourceSerializer.new(PostResource).serialize_to_hash(PostResource.new(post, nil))
```

But when I do the equivalent in 0.10.2, I get the error:

```
undefined method `serialize_to_hash'
 for #<JSONAPI::ResourceSerializer:0x00007fae9ec1cbf0>
```

Looking through the source of `JSONAPI::ResourceSerializer`, [`#object_hash`](https://github.com/cerebris/jsonapi-resources/blob/master/lib/jsonapi/resource_serializer.rb#L144) seems to be the currently-available method that meets that need. The code succeeds for me with the following change:

```
post = Post.find(1)
JSONAPI::ResourceSerializer.new(PostResource).object_hash(PostResource.new(post, nil), nil)
```

(Note that `#object_hash` takes two arguments, so we pass `nil` for the second.)

I would PR this change to the page, but I'm unsure of the correct changes for the rest of the page.

**Comment by soltex1** [comment]
@CodingItWrong Did you find the issue?

**Comment by CodingItWrong** [comment]
@soltex1 I didn't figure out the correct changes to make on the page, no, so I wasn't able to do a PR to update it. As for the issue in the code itself, the alternate approach I mentioned above did work.

**Comment by soltex1** [comment]
Yes, I can confirm that using `object_hash` instead of `serialize_to_hash` works. Not sure why a very simple example doesn't work as mentioned in the documentation. Anyway, thanks for your quick reply! :)

**Comment by soltex1** [comment]
@CodingItWrong Well, it looks like the version [0.10.2](https://github.com/cerebris/jsonapi-resources/blob/v0.10.2/lib/jsonapi/resource_serializer.rb) and even the latest one [0.10.5](https://github.com/cerebris/jsonapi-resources/blob/v0.10.5/lib/jsonapi/resource_serializer.rb) doesn't contain the method `serialize_to_hash`, but the [master](https://github.com/cerebris/jsonapi-resources/blob/master/lib/jsonapi/resource_serializer.rb#L46) does.

The documentation tells us to add `gem 'jsonapi-resources'` to the Gemfile which is going to install the `0.10.5`, and then they use the method `serialize_to_hash` in their examples that doesn't exist on that version.

We might have to install the gem directly from the github by adding this line to the Gemfile:

`gem 'jsonapi-resources', :git => 'https://github.com/cerebris/jsonapi-resources.git'`

Unfortunately, I have a few conflicts with other gems and I wasn't able to test if this works.

**Comment by womanonrails** [comment]
[Here](https://github.com/cerebris/jsonapi-resources/issues/1149) is the issue about this problem on `jsonapi-resources` GitHub.

**Comment by benbonnet** [comment]
@soltex1 can confirm it works with `gem 'jsonapi-resources', :git => 'https://github.com/cerebris/jsonapi-resources.git'`

**Comment by HenrichJones** [comment]
@benbonnet worked for me!

**Comment by Genkilabs** [comment]
For those wanting to use `'~> 0.10.7'` (in my case Rails 5.2.8), you can swap out `serialize_to_hash` with `object_hash` but it requires 2 params. The second can be an empty hash.
For a nice TLDR drop-in, check out @xhocquet solution over on the 1149 issue:
https://github.com/cerebris/jsonapi-resources/issues/1149#issuecomment-935096193
